bolt-sidecar/src/driver.rs

```diff
@@ -35,6 +40,23 @@ pub struct SidecarDriver<C, BLS, ECDSA> {
     mevboost_client: MevBoostClient,
     api_events_rx: mpsc::Receiver<CommitmentEvent>,
     payload_requests_rx: mpsc::Receiver<FetchPayloadRequest>,
+    consensus_clock: Clock<SystemTimeProvider>,
```

Comment:
if we only need the SlotStream from the clock, I would prefer to keep that in the struct instead of the Clock.
It should be of type `SlotStream<something>` but if it's an issue let me know!